lib/Driver/Compilation.cpp

```diff
@@ -810,15 +814,29 @@ namespace driver {
       do {
         using namespace std::placeholders;
         // Ask the TaskQueue to execute.
-        TQ->execute(std::bind(&PerformJobsState::taskBegan, this,
-                              _1, _2),
-                    std::bind(&PerformJobsState::taskFinished, this,
-                              _1, _2, _3, _4, _5),
-                    std::bind(&PerformJobsState::taskSignalled, this,
-                              _1, _2, _3, _4, _5, _6));
-
-        // Returning from TaskQueue::execute should mean either an empty
-        // TaskQueue or a failed subprocess.
+        if (TQ->execute(std::bind(&PerformJobsState::taskBegan, this,
+                                  _1, _2),
+                        std::bind(&PerformJobsState::taskFinished, this,
+                                  _1, _2, _3, _4, _5),
+                        std::bind(&PerformJobsState::taskSignalled, this,
+                                  _1, _2, _3, _4, _5, _6))) {
+          if (Result == EXIT_SUCCESS) {
+            // Error from task queue while Result == EXIT_SUCCESS most likely
+            // means some fork/exec or posix_spawn failed; TaskQueue saw "an
+            // error" at some stage before even calling us with a process exit /
+            // signal (or else a poll failed); unfortunately the task causing it
+            // was dropped on the floor and we have no way to recover it here,
+            // so we report a very poor, generic error.
```

Comment:
Better than nothing, and I'm glad we caught it. Maybe this should be a FIXME, though?

Reply:
Sure